src/track/taglib/trackmetadata_id3v2.cpp

```diff
@@ -83,24 +83,45 @@ inline QString frameToQString(
     return toQString(frame.toString());
 }
 
+bool logWarningIfUnsupportedID3v2Frame(const TagLib::ID3v2::Frame* pFrame) {
+    if (dynamic_cast<const TagLib::ID3v2::UnknownFrame*>(pFrame)) {
+        kLogger.warning()
+                << "ID3v2 frame"
+                << pFrame->frameID().data()
+                << "is not yet supported by TagLib";
+        return true;
+    }
+    // Not an of UnknownFrame, i.e. maybe some other kind of ID3v2 frame
+    return false;
+}
+
+void logWarningAboutUnsupportedOrUnexpectedID3v2Frame(const TagLib::ID3v2::Frame* pFrame) {
+    if (!logWarningIfUnsupportedID3v2Frame(pFrame)) {
+        // Do not crash if the caller unexpectedly passed a nullptr
+        VERIFY_OR_DEBUG_ASSERT(pFrame) {
+            return;
+        }
```

Comment:
If the caller passed a nullptr, we would have never entered this case though. also the assertion seems backwards to me, a nullptr is falsy.

Reply:
Wrong. logWarningIfUnsupportedID3v2Frame() simply returns false and we enter the if arm.

Reply:
My bad, I somehow had my understanding of DEBUG_ASSERTs mixed up.